exercicios/para-casa/sistema_de_alunas.SamanthaTeng.py

```diff
@@ -0,0 +1,176 @@
+                                    "6 - Sair do sistema\n"
+                                    "Opção: "))
+                
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 5).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+            
+        return codigo_opcao
+
+def incluir_nova_aluna():
+    print("Insira os seguintes dados: ")
+    nome = input("Nome da aluna: ") 
+    sobrenome = input ("Sobrenome: ")
+    turma = input("Turma da aluna: ") 
+    lista_presenca = obter_presenca() 
+    nota_participacao = float(input("Participação da aluna: "))
+    notas = obter_notas() 
+    salvar_dados_aluna(nome,sobrenome, turma, notas, lista_presenca, nota_participacao) 
+    return nome
+
+
+def obter_presenca():
+    quantidade_aulas = input("Quantidade de aulas: ") 
+    aulas = [] 
+    
+    for contador in range(int(quantidade_aulas)): 
+        
+        while True:    
+            entrada = input(f"Insira a presença da aula #{contador + 1}: ") 
+            try: 
+                presenca = eval(entrada) 
+                aulas.append(presenca) 
+                break 
+            except NameError: 
+                print("Entrada inválida. Por favor, insira True ou False.")
+   
+    return aulas
+    
+def obter_notas():
+
+    quantidade_notas = input("Quantidade de notas: ") 
+    notas = [] 
+    
+    for contador in range(int(quantidade_notas)): 
+       
+        while True: 
+            entrada = input(f"Insira a nota #{contador + 1}: ") 
+            try: 
+                nota = float(entrada) 
+                notas.append(nota) 
+                break 
+            except ValueError:
+                print("Entrada inválida. Por favor, insira um número válido.")
+                
+    
+    return notas
+
+def salvar_dados_aluna (nome,sobrenome, turma, notas, lista_presenca, nota_participacao):
+    chave = (nome) 
+    dataset[chave] = { 
+        "Sobrenome": sobrenome,
+        "Turma": turma,
+        "Notas": notas,
+        "Presença": lista_presenca,
+        "Participação": nota_participacao
+    }
+    print ("Aluna foi adicionada no sistema com sucesso!")
+
+    
+def consultar_lista_alunas():
+    print ("Os nomes inseridos foram:")
+    for key in dataset.keys():
+        print(key)
+    
+def consultar_faltas_aluna():
+    nome = input("Insira o nome da aluna: ")
+    sobrenome = input("Insira o sobrenome da aluna: ") 
+    nome_completo = dataset[nome]['Sobrenome']
+    if sobrenome != nome_completo:
+        print("Aluna não encontrada")
+        return ValueError
+    nome_completo = nome + ' ' + nome_completo
+    faltas = dataset[nome]['Presença']
+    contador = 0
+    for f in faltas:
+        if not f:
+            contador += 1
+    faltas = str(contador)
+    print('nome completo: ' + nome_completo + ' ' + 'quantidade de faltas: ' + faltas)
```

Comment:
Este código está correto, porém a função count da lista tem o mesmo efeito. Para treino e desenvolvimento inicial é interessante, mas posteriormente, podemos explorar as funções existentes para não precisar implementar as coisas do 0.

**Sugestão:**
```suggestion
    faltas.count(False)
```